src/eureka/S5_lightcurve_fitting/differentiable_models/GPModel.py

```diff
@@ -0,0 +1,278 @@
+
+            # Make the gp object
+            gp = celerite2.GaussianProcess(kernel, mean=0.)
+            gp.compute(self.kernel_inputs[chan][0], yerr=unc_fit)
+
+            # Predict values
+            lcpiece = gp.predict(residuals).eval()
+
+            # Append this channel to the outputs
+            lcfinal = np.append(lcfinal, lcpiece)
+
+        return lcfinal
+
+    def setup_inputs(self):
+        """Setting up kernel inputs as array and standardizing them if asked.
+        
+        For details on the benefits of normalization, see e.g.
+        Evans et al. 2017.
+        """
+        self.kernel_inputs = []
+        for c in range(self.nchannel_fitted):
+            if self.nchannel_fitted > 1:
+                chan = self.fitted_channels[c]
+            else:
+                chan = 0
+
+            kernel_inputs_channel = []
+            for name in self.kernel_input_names:
+                if name == 'time':
+                    x = np.copy(self.time)        
+                else:
+                    # add more input options here
+                    raise ValueError('Currently, only GPs as a function of '
+                                     'time are supported, but you have '
+                                     'specified a GP as a function of '
+                                     f'{name}.')
+
+                if self.multwhite:
+                    x = split([x, ], self.nints, chan)[0]
+
+                if self.normalize:
+                    x = (x-x.mean())/x.std()
+
+                kernel_inputs_channel.append(x)
+            kernel_inputs_channel = np.array(kernel_inputs_channel)
+
+            # Need to reshape if there is only one covariate
+            if len(kernel_inputs_channel.shape) == 1:
+                kernel_inputs_channel = kernel_inputs_channel[np.newaxis]
+
+            self.kernel_inputs.append(kernel_inputs_channel)
+
+    def get_kernel(self, lib, kernel_name, coeffs, k, c=0):
+        """Get individual kernels.
+
+        Parameters
+        ----------
+        lib : module
+            Either the np module for evaluated models or the tt module for
+            gradient-based inference.
+        kernel_name : str
+            The name of the kernel to get.
+        coeffs : list
+            A three-dimensional list of GP hyperparameter coefficients of shape
+            (self.nchannel_fitted, self.nkernels, 2).
+        k : int
+            The kernel number.
+        c : int; optional
+            The channel index, by default 0.
+
+        Returns
+        -------
+        kernel
+            The requested kernel.
+        """
+        # get metric and amplitude for the current kernel and channel
+        amp = lib.exp(coeffs[c][k][0])
+        ls = lib.exp(coeffs[c][k][1])
+
+        if kernel_name == 'Matern32':
+            kernel = celerite2.terms.Matern32Term(sigma=1, rho=ls)
+        else:
+            raise AssertionError('Our celerite2 implementation currently only '
```

Comment:
Should we consider implementing a few other kernels in this PR?

Reply:
The small number of available celerite2 kernels are listed here: https://celerite2.readthedocs.io/en/latest/api/python/#python-recommended

I don't expect that the SHO kernel would be overly useful for our kind of data

Reply:
Ok, thanks!